infra/modules/aurora-db/main.tf

```diff
@@ -0,0 +1,201 @@
+  engine_mode        = "provisioned"
+  database_name      = replace("${var.service_name}", "-", "_")
+  master_username    = "app_usr"
+  master_password    = aws_ssm_parameter.random_db_password.value
+  storage_encrypted  = true
+  # checkov:skip=CKV_AWS_128:Auth decision needs to be ironed out
+  # checkov:skip=CKV_AWS_162:Auth decision needs to be ironed out
+  # iam_database_authentication_enabled = true
+  deletion_protection = true
+  # final_snapshot_identifier = "${var.service_name}-final"
+  skip_final_snapshot = true
+
+
+  serverlessv2_scaling_configuration {
+    max_capacity = 1.0
+    min_capacity = 0.5
+  }
+
+}
+
+resource "aws_rds_cluster_instance" "postgresql-cluster" {
+  cluster_identifier         = aws_rds_cluster.postgresql.id
+  instance_class             = "db.serverless"
+  engine                     = aws_rds_cluster.postgresql.engine
+  engine_version             = aws_rds_cluster.postgresql.engine_version
+  auto_minor_version_upgrade = true
+  monitoring_role_arn        = aws_iam_role.rds_enhanced_monitoring.arn
+  monitoring_interval        = 30
+}
+
+resource "random_password" "random_db_password" {
+  length           = 48
+  special          = true
+  min_special      = 6
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+
+resource "aws_ssm_parameter" "random_db_password" {
+  name  = "/metadata/db/admin-password"
+  type  = "SecureString"
+  value = random_password.random_db_password.result
+}
+
+################################################################################
+# Backup Configuration
+################################################################################
+
+resource "aws_backup_plan" "postgresql" {
+  name = "${var.service_name}_backup_plan"
+
+  rule {
+    rule_name         = "${var.service_name}_backup_rule"
+    target_vault_name = "${var.service_name}-vault"
+    schedule          = "cron(0 12 ? * SUN *)"
+  }
+}
+
+# KMS Key for the vault
+# This key was created by AWS by default alongside the vault
+data "aws_kms_key" "postgresql" {
+  key_id = "alias/aws/backup"
+}
+# create backup vault
+resource "aws_backup_vault" "postgresql" {
+  name        = "${var.service_name}-vault"
+  kms_key_arn = data.aws_kms_key.postgresql.arn
+}
+
+# create IAM role
+resource "aws_iam_role" "postgresql_backup" {
+  name_prefix        = "aurora-backup-"
+  assume_role_policy = data.aws_iam_policy_document.postgresql_backup.json
+}
+
+resource "aws_iam_role_policy_attachment" "postgresql_backup" {
+  role       = aws_iam_role.postgresql_backup.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
+}
+
+data "aws_iam_policy_document" "postgresql_backup" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
+    }
+  }
+}
+# backup selection
+resource "aws_backup_selection" "postgresql_backup" {
```

Comment:
nit, could we put this resource near the top of this section right after the backup plan? i think the backup plan and the backup selection are the most important resources for a reader to understand what's being backed up, and the iam stuff is secondary/supporting.